mm.lic

```diff
@@ -0,0 +1,94 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#mm
+=end
+custom_require.call(%w[common-arcana events])
+
+class MoonMagery
+  include DRCA
+  def initialize
+    args = get_args
+    @settings = get_settings
+
+    if args.locate
+      locate?(args.person)
+    elsif args.rift
+      rift(args.person)
+    end
+    Flags.delete('locate-fail')
+  end
+
+  def get_args
+    arg_defs = [
+      [
+        { name: 'locate', regex: /locate/i },
+        { name: 'person', regex: /\w+/, description: 'Name of person to locate' }
+      ],
+      [
+        { name: 'rift', regex: /rift/i },
+        { name: 'person', regex: /\w+/, description: 'Name of person to rift' }
+      ]
+    ]
+
+    parse_args(arg_defs)
+  end
+
+  def rift(person)
+    person = person.capitalize
+    check_moonwatch
+    unless UserVars.moons.any? { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 0 }
+      echo 'No moons available to cast Rift'
+      return
+    end
+    data = if @settings.waggle_sets['rs']
+             @settings.waggle_sets['rs'].values.first
+           else
+             echo "Using discern values for RS. Make a waggle_set called 'rs' to avoid this in the future"
```

Comment:
Do you think users would expect a waggle_set to be used by a script other than `buff.lic`?

Reply:
I wont speak for all users but there is precedent for it. Pick, astrology, zombie are three set names I can think of

Reply:
The difference I'm seeing is that `pick` calls `buff` and just passes in the name of the waggle set. However, this uses the waggle set data directly. You're right that `astrology` does the same as `mm` though. I'm torn on this, so I'm willing to leave it alone and see what patterns emerge...